hypothesis-python/src/hypothesis/extra/django/_fields.py

```diff
@@ -183,20 +247,38 @@ def from_field(field):
     than a Field *subtype*, so that it has access to instance attributes
     such as string length and validators.
     """
-    check_type(dm.Field, field, "field")
-    if field.choices:
+    check_type((dm.Field, df.Field), field, "field")
+    if getattr(field, "choices", False):
         choices = []  # type: list
         for value, name_or_optgroup in field.choices:
             if isinstance(name_or_optgroup, (list, tuple)):
                 choices.extend(key for key, _ in name_or_optgroup)
             else:
                 choices.append(value)
+        # form fields automatically include an empty choice, strip it out
+        if "" in choices:
```

Comment:
Pretty sure the problem is that in Python 2, `u""` is different to `""`.  I'd try adding the `u` prefix here, and of course also in the `.remove` call.